ICSharpCode.Decompiler/CSharp/SequencePointBuilder.cs

```diff
@@ -113,7 +113,13 @@ public override void VisitBlockStatement(BlockStatement blockStatement)
 			ILInstruction blockContainer = blockStatement.Annotations.OfType<ILInstruction>().FirstOrDefault();
 			if (blockContainer != null) {
 				StartSequencePoint(blockStatement.LBraceToken);
-				int intervalStart = blockContainer.ILRanges.First().Start;
+				int intervalStart;
+				if (blockContainer.Parent is TryCatchHandler handler && !handler.ExceptionSpecifierILRange.IsEmpty) {
```

Comment:
This condition is true for both the filter and the handler blocks.
We really need tests that both the cases with and without filter emit sequence points as expected.

Reply:
`catch` blocks with a `when` clause do not have an `ExceptionSpecifierILRange`. The sequence point is on the when clause and not on the catch specifier.

This is how Visual Studio shows the breakpoints:
![image](https://user-images.githubusercontent.com/344204/79645187-0405c100-81ae-11ea-8d60-48a17c694a1f.png)